src/System.Management.Automation/engine/Utils.cs

```diff
@@ -1260,6 +1260,17 @@ internal static bool PathIsUnc(string path)
 #if UNIX
             return false;
 #else
+            if (!path.StartsWith('\\'))
+            {
+                return false;
+            }
+
+            // handle special cases like \\wsl$\ubuntu which isn't a UNC path, but we can say it is so the filesystemprovider can use it
+            if (path.StartsWith(@"\\wsl$", StringComparison.OrdinalIgnoreCase))
+            {
+                return true;
+            }
+
             Uri uri;
             return !string.IsNullOrEmpty(path) && Uri.TryCreate(path, UriKind.Absolute, out uri) && uri.IsUnc;
```

Comment:
The `IsNullOrEmpty` check should be moved to the top before calling `StartsWith` on `path`.

Reply:
Will update